src/odil/DataSet.cpp

```diff
@@ -432,7 +432,7 @@ ::size(Tag const & tag) const
     ElementMap::const_iterator const it = this->_elements.find(tag);
     if(it == this->_elements.end())
     {
-        throw Exception("No such element");
+        throw Exception("No such element "+ tag.get_name());
```

Comment:
`Tag.get_name` may throw (e.g. for private tags): the safest way would be to use `operator std::string()` instead.